app/javascript/packages/verify-flow/steps/personal-key/download-button.tsx

```diff
@@ -0,0 +1,41 @@
+import type { MouseEventHandler } from 'react';
+import { Button } from '@18f/identity-components';
+import type { ButtonProps } from '@18f/identity-components';
+
+interface DownloadButtonProps {
+  /**
+   * Content of the downloaded file.
+   */
+  content: string;
+
+  /**
+   * File name to use for downloaded file.
+   */
+  fileName: string;
+}
+
+function DownloadButton({ content, fileName, ...buttonProps }: DownloadButtonProps & ButtonProps) {
+  const download: MouseEventHandler = (event) => {
+    buttonProps.onClick?.(event);
+
+    if ('msSaveBlob' in window.navigator) {
+      event.preventDefault();
+
+      const filename = (event.target as HTMLButtonElement).getAttribute('download');
```

Comment:
> Why not just grab `download` as a prop of the `DownloadButton` component?

That's a good point. I copy-pasted this logic, and the idea hadn't occurred to me, but that's much simpler. 👍

Reply:
Updated in b333339ed.

Also, related to the other comment, with typed interface of `saveBlob`, the `getAttribute` would fail type-check, since it's possibly `null`. Reading from the prop helps avoid this.